frontend/lib/src/components/widgets/DataFrame/columns/utils.ts

```diff
@@ -472,30 +472,59 @@ export function formatNumber(
       }
 
       return numbro(value).format({
-        thousandSeparated: true,
+        thousandSeparated: false,
         mantissa: maxPrecision,
         trimMantissa: false,
       })
     }
 
     // Use a default format if no precision is given
     return numbro(value).format({
-      thousandSeparated: true,
+      thousandSeparated: false,
       mantissa: determineDefaultMantissa(value),
       trimMantissa: true,
     })
   }
 
-  if (format === "percent") {
+  if (format === "plain") {
+    return numbro(value).format({
+      thousandSeparated: false,
+      // Use a large mantissa to avoid cutting off decimals
+      mantissa: 20,
+      trimMantissa: true,
+    })
+  } else if (format === "localized") {
+    return new Intl.NumberFormat().format(value)
+  } else if (format === "percent") {
     return new Intl.NumberFormat(undefined, {
       style: "percent",
-      minimumFractionDigits: 2,
+      minimumFractionDigits: 0,
+      maximumFractionDigits: 2,
+    }).format(value)
+  } else if (format === "dollar") {
```

Comment:
would it make sense to use the same currency abbreviation as used in the `Intl` library? So instead of `dollar -> USD` etc.? Maybe then you could also do something like `if (format typeof Intl.currency)` and have only a single if block for all currencies; this would make it very easy to add other currencies in the future. I have not looked in depth whether its possible though.

Reply:
Yeah, good point. I thought about unlocking it for a lot more currencies, but we would still need to check against a list of supported currencies since the fallback is to pass it into `sprintf`. Also, if we support USD directly we should probably use the normal symbol display and not the `narrowSymbol` (`US$`). I think we can leave `dollar` in as a more generic way to activate the dollar sign, which can in theory be used for all currencies that use the `$` sign. My suggestion:

We leave it as is and I will create a new feature request to support more currencies. If this gets enough upvotes, we will add support to directly use the currencies via `USD`, `EUR`, ... and use the normal symbol (not narrow).